src/server.cpp

```diff
@@ -3870,8 +3870,13 @@ v3f Server::findSpawnPos()
 {
 	ServerMap &map = m_env->getServerMap();
 	v3f nodeposf;
-	if (g_settings->getV3FNoEx("static_spawnpoint", nodeposf))
-		return nodeposf * BS;
+
+    std::optional<v3f> staticSpawnPoint;
+	if (g_settings->getV3FNoEx("static_spawnpoint", staticSpawnPoint) && staticSpawnPoint.has_value())
+    {
+        nodeposf = staticSpawnPoint.value();
```

Comment:
```suggestion
       return *staticSpawnPoint * BS;
```

then get rid of the `v3f nodeposf;` above (you will have to introduce a `v3f` a bit below, basically you're making the scope more limited)